### PR TITLE
PLT-282: Build the Agda doc site in CI

### PR DESCRIPTION
**Pull request by michaelpj**
The README link won't work until this is merged, but it should be fine after that.

Definitely some improvements to be made to the site (it's unstyled!), but I'll leave those for later.

### CODE REVIEW

File: nix/cells/plutus/packages/plutus-metatheory-site.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
+{ inputs, cell }:
+
+let
+  # Doing this in two derivations so the call to agda is cached, since
+  # that's very slow. Makes it easier to iterate on the site build.
+  plutus-metatheory-agda-html = cell.library.pkgs.stdenv.mkDerivation {
+    name = "plutus-metatheory-doc";
+    # Not doing this seems to lead to spurious rebuilds, I think because it
```
Comment (michaelpj) [inline nix/cells/plutus/packages/plutus-metatheory-site.nix:8]: FYI @zeme-iohk this is the thing I slacked you about. Seems suspicious.

Comment (zeme-iohk) [reply nix/cells/plutus/packages/plutus-metatheory-site.nix:8]: That's odd, we use the pattern `inputs.self + /sub/dir` in many other places. 🤔

**Review by mjaskelioff — APPROVED** [review]

File: README.adoc [hunk 2]
```diff
@@ -44,6 +44,8 @@ After setting it up you should just be able to depend on the `plutus` packages a
 
 The main documentation is located https://plutus.readthedocs.io/en/latest/[here].
 
+The latest documentation for the metatheory can be found https://ci.zw3rk.com/job/input-output-hk-plutus/master/x86_64-linux.plutus-metatheory-site/latest/download/1[here].
```
[the remark this answers is not in the document]
Comment (zeme-iohk) [reply README.adoc:47]: It needs to be pushed to master once first.

**Comment by jmchapman** [comment]
The site seems functional - thank you!!!

 But I thought it was perhaps missing some CSS but it seems the parts of the html are missing:

Compare:
https://input-output-hk.github.io/plutus-metatheory/

and 

https://ci.zw3rk.com/build/1284320/download/1/plutus-metatheory-site/#normal-types

**Comment by jmchapman** [comment]
i.e. viewing source on the above page shows that the file starts with `<pre>` not `<html>`

**Comment by mjaskelioff** [comment]
I think the `.lagda.md` files are not being generated at all. For example 
```
https://ci.zw3rk.com/build/1284320/download/1/plutus-metatheory-site/Algorithmic.CEKV.html
```
Gives a 404.

**Comment by michaelpj** [comment]
Okay, are you sure you didn't have some extra steps when you generated that? Like a Jekyll config file?

**Comment by michaelpj** [comment]
I'm just running the command you gave me :D

**Pull request merged**